### DISCUSSION TITLE
how to open the aio threads in apisix

### DISCUSSION THREAD

**Discussion by Lionelpang** [apache/apisix#5334] [General]
how to config the apisix open the aio threads for the nginx。

**Comment by tokers** [comment]
With the help of `main_configuration_snippet` field in `config.yaml`, you can enable any Nginx directive.  But you may have to re-compile the openresty and enable the file aio (`--with-file-aio`) by yourself, you may check out http://apisix.apache.org/docs/apisix/how-to-build/#step-6-build-openresty-for-apache-apisix to learn how to build your own OpenResty for APISIX.
